### ISSUE TITLE
Translation/generation scripts?

### ISSUE THREAD

**Issue by kavehtp** [facebookresearch/MIXER#2]
Is there any test script in the repo for translating a text file given a model? Does it support beam search?

**Comment by ranzato** [comment]
- test script in the repo for translating a text file given a model?
  Yes, check how we perform evaluation computing BLEU. The entry point is at:
  https://github.com/facebookresearch/MIXER/blob/master/main.lua#L163
  which goes to:
  https://github.com/facebookresearch/MIXER/blob/master/Trainer.lua#L208
  and
  https://github.com/facebookresearch/MIXER/blob/master/Mixer.lua#L333
  This assumes that you have already tokenized data and have a trained model. You could easily extend this to take raw text and do tokenization on the fly using the dictionary. 
- Does it support beam search?
  No, the current code release only supports greedy generation both at training and test time.

**Comment by Maggione** [comment]
I have ran an LSTM with the default parameter setting used to generate MIXER's entry for machine translation, by the command "th -I main.lua", however, I got BLEU score like this:
[[ loading previously trained model ./backups/25/model_8 ]] 	
Length of generations:	137996	Length of ground truth:	112571	Brevity penalty:	1	Corpus level BLEU:	0.12249842847183	Sentence level BLEU:	0.083373183739123	Total number of sentences:	6778	
Is it right? Why the score is so low? What does it mean?

**Comment by ranzato** [comment]
The reason why is not good is because you are evaluating a model that is not fully trained. At that stage, the model has only been trained with standard token-level cross-entropy loss, and not even for long enough. You need to keep training for longer and evaluate at the very end. The fix is to let it run till the end.

**Comment by Maggione** [comment]
Thank you for reply! I have trained the model with standard token-level cross-entropy loss for 25 epoch and the log is showed as blow. The BLEU score is still low. I think it may be the baseline BLEU score. 
I wonder whether the calculation of the scores of "Corpus level BLEU" the same as the BLEU scores showed in your paper?
 
Epoch: 25. Training time: 1255.34s. WordsXE/s: 2090.34, WordsRF/s: 30.94
Training: Ent: 2.50410 || Ppl: 5.67296 || Avg. reward/token: 0.000 || Cum. reward error: 0.000 || Nsamples Xent: 2624082 || Nsamples Rf: 38842
Validation: Ent: 4.10220 || Ppl: 17.17453
Decreasing the learning rate to 0.046513607872276Length of generations: 134595  Length of ground truth: 112571  Brevity penalty:    1   Corpus level BLEU:  0.15807637295071    Sentence level BLEU:    0.1210212620008 Total number of sentences:  6778 

Thank you again!
